src/main/java/com/rarchives/ripme/ripper/rippers/AllporncomicRipper.java

```diff
@@ -0,0 +1,63 @@
+package com.rarchives.ripme.ripper.rippers;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import com.rarchives.ripme.ripper.AbstractHTMLRipper;
+import com.rarchives.ripme.utils.Http;
+
+public class AllporncomicRipper extends AbstractHTMLRipper {
+
+    public AllporncomicRipper(URL url) throws IOException {
+        super(url);
+    }
+
+    @Override
+    public String getHost() {
+        return "allporncomic";
+    }
+
+    @Override
+    public String getDomain() {
+        return "allporncomic.com";
+    }
+
+    @Override
+    public String getGID(URL url) throws MalformedURLException {
+        Pattern p = Pattern.compile("https?://allporncomic.com/porncomic/([a-zA-Z0-9_-]+)/([a-zA-Z0-9_-]+)");
+        Matcher m = p.matcher(url.toExternalForm());
+        if (m.matches()) {
+            return m.group(1) + "_" + m.group(2);
+        }
+        throw new MalformedURLException("Expected cfake URL format: " +
+                "cfake.com/picture/MODEL/ID - got " + url + " instead");
```

Comment:
You forgot to edit this part when copying and pasting from the original values.

Reply:
Didn't realise you were cyian-1756. I'm sure you were getting there.

Reply:
Totally missed that, thanks for pointing it out

Reply:
Not a problem at all. Not trying to snatch the pull request or anything, but if you've time maybe could you check #1362? Without that the ripper I've added doesn't function on half of the pages. I wanted to send a PM but I guess GitHub doesn't have that.